care/facility/api/serializers/shifting.py

```diff
@@ -324,7 +324,18 @@ def update(self, instance, validated_data):
             "status" in validated_data
             and validated_data["status"] == REVERSE_SHIFTING_STATUS_CHOICES["COMPLETED"]
         ):
-            discharge_patient(instance.patient)
+            # Staff from the destination must not be able to mark the shift as complete.
+            destination_facility = (
+                validated_data.get("assigned_facility") or instance.assigned_facility
+            )
+            if destination_facility and destination_facility.users.contains(user):
```

Comment:
if a user is part of the destination and receiving facility this might become a problem.